src/gui/IconDownloader.cpp

```diff
@@ -114,7 +103,7 @@ void IconDownloader::setUrl(const QString& entryUrl)
     // Determine the second-level domain, if available
     QString secondLevelDomain;
     if (!hostIsIp) {
-        secondLevelDomain = getSecondLevelDomain(url);
+        secondLevelDomain = Tools::baseDomain(url);
```

Comment:
This is actually wrong, we should just "chop up" a domain level (ie, my.example.com -> example.com) even if it means we end up with just a TLD (which will not resolve anyway). The current call here will always send us to the base domain potentially bypassing a level (level2.level1.domain.com -> domain.com, instead of level1.domain.com).

Reply:
I think we should be using Mozilla's public suffix list here instead of naively chopping up domains. It requires the loading of another resource and some parsing logic for the list, but everything else is somewhat unsafe.

Reply:
@droidmonkey: My intent here was to keep the same behaviour. If you are saying the current behaviour is incorrect, I suggest fixing that in a separate PR. Or are you saying I am (accidentally) breaking the behaviour here?

@phoerious: The "clean" way to do this would be to use something like [libpsl](https://github.com/rockdaboot/libpsl). I just wanted to avoid adding another external dependency like that. And I feel that my current "unconventional" solution should be okay given that we have solid unit tests for this.

Reply:
I am saying you are breaking the behavior.

Reply:
> we should just "chop up" a domain level (ie, my.example.com -> example.com) even if it means we end up with just a TLD (which will not resolve anyway)

Not sure I understand. Both before and after my change, we have `my.example.com` → `example.com` and `example.com` → `example.com`. This seems reasonable to me.

> The current call here will always send us to the base domain potentially bypassing a level (level2.level1.domain.com -> domain.com, instead of level1.domain.com).

That is also the old behaviour, even according to the source code comment for `getSecondLevelDomain()`.

The only case where I can think of a difference in behaviour is with an invalid URL like "`foo`", where `getSecondLevelDomain()` would presumably crash/throw in `parts.takeLast()` because `parts` is empty, whereas `Tools::baseDomain()` handles it fine as proven in the unit tests.

Reply:
I have made replacement for this using the Public Suffix List. I'm making a separate PR of it.

Reply:
great, I really want to merge this one. I'll merge yours first though.

Reply:
Marking this resolved, since my latest push has deleted the relevant commit in favour of #9935.